src/MeshManager.cpp

```diff
@@ -268,6 +286,104 @@ namespace blitzdg {
+                index_type v1 = localVertNum[f][0];
+                index_type v2 = localVertNum[f][1];
+
+                index_type v1Global = E2V(k*NumFaces + v1);
+                index_type v2Global = E2V(k*NumFaces + v2);
+
+                // Entry for v1Global.
+                VToF.rowInds(nnz) = v1Global;
+                VToF.elems(nnz) = 1.0;
+                ++nnz;
+
+                // Entry for v2Global
+                VToF.rowInds(nnz) = v2Global;
+                VToF.elems(nnz) = 1.0;
+                ++nnz;
+
+                ++globalFaceNum;
+            }
+        }
+        VToF.colPtrs(totalFaces) = nnz;
+        CSCMat FToF = multiply(transpose(VToF), VToF);
+
+        // Count the number of face-to-face connections.
+        index_type connectionsCount = 0;
+        for (index_type j = 0; j < totalFaces; ++j) {
+            for (index_type k = FToF.colPtrs(j); k < FToF.colPtrs(j + 1); ++k) {
+                index_type i = FToF.rowInds(k);
+                if (i != j && std::abs(FToF.elems(k) - 2.0) < NodeTol) {
+                    ++connectionsCount;
+                }
+            }
+        }
+
+        index_vector_type e1(connectionsCount);
+        index_vector_type e2(connectionsCount);
+        index_vector_type f1(connectionsCount);
+        index_vector_type f2(connectionsCount);
+
+        f1 = 0;
+        f2 = 0;
+        e1 = 0;
+        e2 = 0;
+
+        // store the faces in each connection.
+        index_type c = 0;
+        for (index_type j = 0; j < totalFaces; ++j) {
+            for (index_type k = FToF.colPtrs(j); k < FToF.colPtrs(j + 1); ++k) {
+                index_type i = FToF.rowInds(k);
+                if (i != j && std::abs(FToF.elems(k) - 2.0) < NodeTol) {
+                    f1(c) = i;
+                    f2(c) = j;
+                    ++c;
+                }
+            }
+        }
+
+        // Convert from global face number to element number with local face number.
+        e1 = floor(f1 / NumFaces);
+        f1 = (f1 % NumFaces);
+        e2 = floor(f2 / NumFaces);
+        f2 = (f2 % NumFaces);
+
+        // Build connectivity matrices.
+        index_vector_type& E2E = *EToE;
+        index_vector_type& E2F = *EToF;
+
+        // Populate arrays with self-connection defaults.
+        for (index_type k = 0; k < NumElements; ++k) {
+            for (index_type f = 0; f < NumFaces; ++f) {
+                E2E(k*NumFaces + f) = k;
+                E2F(k*NumFaces + f) = f;
+            }
+        }
+
+        for (index_type i = 0; i < connectionsCount; ++i) {
+            index_type ee1 = e1(i);
+            index_type ee2 = e2(i);
+            index_type ff1 = f1(i);
+            index_type ff2 = f2(i);
+            E2E(ee1*NumFaces + ff1) = ee2;
+            E2F(ee1*NumFaces + ff1) = ff2;
+        }
+    }
+
     void MeshManager::partitionMesh(index_type numPartitions) {
         unique_ptr<index_type[]> eptr(new index_type[NumElements + 1]);
```

Comment:
This is kind of wierd, you don't need a unique_ptr here. Just use `index_vector_type eptr(NumElements + 1)`.